### DISCUSSION TITLE
Allow Generic for `featureName` for `StoreModule.forFeature()`

### DISCUSSION THREAD

**Discussion by ZackDeRose** [ngrx/platform#3084] [Ideas]
## Describe any alternatives/workarounds you're currently using

Had a task to recently lock down the accepted ngrx feature names to a specific set of names.

To accomplish, we created a typed shim for the `StoreModule` that overrode the `forFeature()` method to give a proper type to the featureName param.

Adding another generic to this method (that defaults to any string) should solve this, without causing breaking changes.

## Other information:

Discussed this with @brandonroberts, who asked me to open this issue!

## If accepted, I would be willing to submit a PR for this feature

[x] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

**Comment by brandonroberts** [comment]
Will you add some sample code of what you see the changes enabling?
